buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -343,6 +343,11 @@ class ClusterFormationTasks {
             // this will also allow new and old nodes in the BWC case to become the master
             esConfig['discovery.initial_state_timeout'] = '0s'
         }
+        if (esConfig.containsKey('discovery.zen.master_election.wait_for_joins_timeout') == false) {
+            // We use 30s as the default timeout in rest requests waiting for cluster formation
```

Comment:
can you extend the explanation here to say why we're doing this? See for example https://github.com/elastic/elasticsearch/pull/27332

Reply:
Sure will do before merging :) Thanks for taking a look!